File: src/main/java/com/beat/domain/performance/application/dto/makerPerformance/MakerPerformanceResponse.java
package com.beat.domain.performance.application.dto.makerPerformance;

import java.util.List;

public record MakerPerformanceResponse(
	Long userId,
	List<MakerPerformanceDetailResponse> performances
) {
	public static MakerPerformanceResponse of(
		Long userId,
		List<MakerPerformanceDetailResponse> performances) {
		return new MakerPerformanceResponse(userId, performances);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/MakerPerformanceResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.maker.query.MakerPerformanceListResult

@ConsistentCopyVisibility
data class MakerPerformanceResponse private constructor(
    val userId: Long?,
    val performances: List<MakerPerformanceDetailResponse>,
) {
    companion object {
        fun from(result: MakerPerformanceListResult): MakerPerformanceResponse =
            MakerPerformanceResponse(result.userId, result.performances.map(MakerPerformanceDetailResponse::from))
    }
}
